lib/blocks.php

```diff
@@ -58,6 +58,72 @@ function gutenberg_parse_blocks( $content ) {
 }
 
 /**
+ * Given an array of parsed blocks, returns content string.
+ *
+ * @since 1.3.0
+ *
+ * @param array $blocks Parsed blocks.
+ *
+ * @return string Content string.
+ */
+function gutenberg_serialize_blocks( $blocks ) {
+	return implode( '', array_map( 'gutenberg_serialize_block', $blocks ) );
+}
+
+/**
+ * Given a parsed block, returns content string.
+ *
+ * @since 1.3.0
+ *
+ * @param array $block Parsed block.
+ *
+ * @return string Content string.
+ */
+function gutenberg_serialize_block( $block ) {
+	// Return content of unknown block verbatim.
+	if ( ! isset( $block['blockName'] ) ) {
+		return $block['rawContent'];
+	}
+
+	// Custom formatting for specific block types.
+	if ( 'core/more' === $block['blockName'] ) {
+		$content = '<!--more';
+		if ( ! empty( $block['attrs']['customText'] ) ) {
+			$content .= ' ' . $block['attrs']['customText'];
+		}
+
+		$content .= '-->';
+		if ( ! empty( $block['attrs']['noTeaser'] ) ) {
+			$content .= "\n" . '<!--noteaser-->';
+		}
+
+		return $content;
+	}
+
+	// For standard blocks, return with comment-delimited wrapper.
+	$content = '<!-- wp:' . $block['blockName'] . ' ';
+
+	if ( ! empty( $block['attrs'] ) ) {
+		$attrs_json = json_encode( $block['attrs'] );
+
+		// In PHP 5.4+, we would pass the `JSON_UNESCAPED_SLASHES` option to
+		// `json_encode`. To support older versions, we must apply manually.
```

Comment:
Nice comment!

Also, check out the JS serializer to see the other transforms we make in order to safeguard the attributes

https://github.com/WordPress/gutenberg/blob/e7f12238d84abab88afec915c1b94f7724bc88c8/blocks/api/serializer.js#L120

Reply:
>Also, check out the JS serializer to see the other transforms we make in order to safeguard the attributes

Ah, right, will make a pass to port over this behavior and equivalent tests.